### ISSUE TITLE
add passage identifier to elements of GetPassage response

### ISSUE THREAD

**Issue by balmas** [cite-architecture/cts_spec#13]
It would be nice if there was a standard way for a GetPassage response to identify the passage id of each returned element when the request is for a range.

Although it's possible for a client to reverse-calculate the passage identifiers of the response by parsing the TextInventory and applying the xpath to the returned elements, since the CTS implementation will have to already have done that in some fashion or another, it would be more efficient for it to insert this information on the wrapping passage elements, perhaps as a cts-namespaced attribute. 

Having a standard approach to this would facilitate interoperability of client code that deals with CTS passages.

**Comment by neelsmith** [comment]
I like this idea.

**Comment by michaelgursky** [comment]
Hi Bridget,

In my implementation, I can grab non-leaf nodes in a passage without
traversing them.  If the paths to the start and end of the passage diverge,
it's not necessary to examine nodes in between the start/end siblings at
the highest point of divergence.  I could insert ids, but it would slow
down the retrieval, perhaps significantly.  It would be nice if this were
optional (on the part of the server).

Also, sorry I haven't had much time to work on this recently.  Things have
gotten busy at work.

Mike

**Comment by neelsmith** [comment]
I don't see this is a specification issue -- note that any well-formed contents of the <passage> element will validate against the schema.

Which is to say, Bridget is free to wrap citable notes in some kind of identifier in her own namespace, and Michael is free not to.

**Comment by balmas** [comment]
I agree it could be optional but it would be nice if there was a standard way to specify it when used.

**Comment by neelsmith** [comment]
yes.

just off the top of my head, are you thinking of something like:

<cts:passage n="RANGEURN">

<cts:node n="NODEURN">

notactsnamespace:elementText content./notactsnamespace:element

</ctsnode>

/cts:passage

where cts:node is an optional element?

**Comment by balmas** [comment]
I was actually just thinking of adding an attribute on the element that contains the passage.

e.g. if you have `<div n="1" type="book">....</div>`

then the service could maybe insert a cts:cite attribute
`<div n='1' cts:cite='1'></div>`

Of course I guess this might break the TEI validation but so would a wrapping node.

**Comment by neelsmith** [comment]
That assumes the the contents of cts:passage will always be XML, but pure text content is also valid.  Wrapping an additional node would leave open possiblitieles like

<cts:passage n="RANGE">
<cts:node n="NODEURN">Text content/cts:node
/cts:passage

**Comment by balmas** [comment]
that's a valid point.  I would be happy with either approach I think.
